ner_v1/detectors/textual/name/name_detection.py

```diff
@@ -0,0 +1,170 @@
+        and not present in the templates.
+        :param text:
+                Example text= My name is yash modi
+        :return: [{first_name: "yash", middle_name: None, last_name: "modi"}], [ "yash modi"]
+        """
+
+        entity_value, original_text = [], []
+        pos_tagger_object = POS()
+        pattern1 = re.compile(r"name\s*(is|)\s*([\w\s]+)")
+        pattern2 = re.compile(r"myself\s+([\w\s]+)")
+        name_tokens = text.split(' ')
+        tagged_names = pos_tagger_object.tag(name_tokens)
+        pattern1_match = pattern1.findall(text)
+        pattern2_match = pattern2.findall(text)
+
+        is_question = [word[0] for word in tagged_names if word[1].startswith('WR') or
+                       word[1].startswith('WP')]
+        if is_question:
+            return entity_value, original_text
+
+        if pattern1_match:
+            entity_value, original_text = self.get_format_name(pattern1_match[0][1].split())
+
+        elif pattern2_match:
+            entity_value, original_text = self.get_format_name(pattern2_match[0].split())
+
+        else:
+            pos_words = [word[0] for word in tagged_names if word[1].startswith('NN') or
+                         word[1].startswith('JJ')]
+            if pos_words:
+                entity_value, original_text = self.get_format_name(pos_words)
+
+        return entity_value, original_text
+
+    def detect_entity(self, text, bot_message):
+        """
+        Takes text and and returns names
+        :param bot_message: previous botmessage
+        :param text: The original text
+            Example:
+                text = My name is yash modi
+        :return:
+        [{first_name: "yash", middle_name: None, last_name: "modi"}], [ "yash modi"]
+
+        """
+        if not self.context_check_botmessage(bot_message):
+            return [], []
+        self.text = text
+        self.tagged_text = self.text
+        text_detection_result = self.text_detection_name()
+        replaced_text = self.replace_detected_text(text_detection_result)
+        entity_value, original_text = self.detect_name_entity(replaced_text)
+
+        if not entity_value:
+            entity_value, original_text = self.get_name_using_pos_tagger(text)
+
+        return entity_value, original_text
+
+    def replace_detected_text(self, text_detection_result):
+        """
+        Replaces the detected name by _<name>_
+        :param text_detection_result: list of detected names from TextDetection
+        :return: tokenized list with detected names replaced by _ _
+        """
+        replaced_text = Tokenizer().tokenize(self.text.lower())
+        for i in (text_detection_result[1]):
+            for j in range(len(replaced_text)):
+                replaced_text[j] = replaced_text[j].replace(i, "_" + i + "_")
+
+        return replaced_text
+
+    def detect_name_entity(self, replaced_text):
+        """
+        Forms a dictionary of the names
```

Comment:
Elaborate docstrings.. 'Forms a dictionary of the names' is as good as not writing docs at all.